### PR TITLE
Declarations applied to telescopes

### PR DESCRIPTION
**Pull request by cangiuli**
For example:
```
def path (A : type) (x : A) (y : A) : type = {
  pathd {_ => A} x y
}
```

### CODE REVIEW

**Comment by cangiuli** [comment]
I started to change some of the tests to this style, but realized I should take a poll about indentation style before I do so. How did everyone feel about how we indented long types in redtt:

```
def square
  (A : type)
  (M N : 𝕀 → A)
  (O : path A (M 0) (N 0))
  (P : path A (M 1) (N 1))
  : type
  =
  ...
```

**Comment by jonsterling** [comment]
I hate the old indentation, let me think about it

**Comment by ivoysey** [comment]
i do not hate the old indentation. i probably won't hate whatever jon comes up with after thinking about it, but i guess we'll see. i think my rough standard for Agda has been
```
identifier : {implicit} {things} {all} {on} {one} {line} →
             interesting things →
             that i'm like to induct on →
             on separate lines under the colon →
             result i'm going to prove
```

**Comment by jonsterling** [comment]
So, the thing that I don't like about your proposal @ivoysey is that I basically prefer to rule out any kind of vertical alignment that depends on some identifier's width.

**Comment by ivoysey** [comment]
yeah, i think i could get away with it in a lot my agda stuff because often there was really only one identifier per file, or a pair of them that were the same except for ending in `-syn` versus `-ana` because of a bidirectional type system, so they coincidentally lined up.

my preference in general is that i like to go tall not long, i.e. use vertical space and not horizontal. i find long lines with more than one thing going on in them very hard to parse. (the long line of implicit idenitifers in my agda standard is "one thing" in the sense that it's "i dunno man i don't care bring all these letters in to scope" that you never really read.)

i do not feel *hugely* passionately about these things but i could imagine something that would bother me.

**Comment by ivoysey** [comment]
(also, doing a little bit of a deep dive into some agda i've written, i am a little shocked at how inconsistent i have been even on repos where i'm the only author. so. maybe i shouldn't talk.)

**Comment by jonsterling** [comment]
@ivoysey been there hahaha

File: src/lib/Driver.ml [hunk 1]
```diff
@@ -46,26 +46,26 @@ let pp_message fmt =
 
 module EM = ElabBasics
 
-let elaborate_typed_term name tp tm =
+let elaborate_typed_term name (args : ConcreteSyntaxData.cell list) tp tm =
```
Comment (jonsterling) [inline src/lib/Driver.ml:49]: change to `CS.cell`
